content/blog/handling-technical-ids-in-gherkin-with-specflow.md

```diff
@@ -0,0 +1,154 @@
+---
+title: "Handling technical id's in Gherkin with SpecFlow"
+date: 2020-06-27T00:00:00+02:00
+image: "images/handling-technical-ids-in-gherkin-with-specflow.jpg"
+tags: [ "Gherkin", "SpecFlow", "Specification by Example", "ATDD", "BDD", "Test Automation", "Cleaner Code" ]
+summary: "Gherkin scenarios in Specification by Example are used to describe the functional requirements of your software. They should be readable for the team and also for the business people that use the software. Technical id's don't have a place here. So what to do when your code requires a technical id?"
+draft: true
+---
+
+When you use Specification by Example with the Gherkin syntax and automate your scenarios with SpecFlow, you're bound to encounter situations where you'll need a technical id. For example to stub data that's retrieved from a repository or external service.
+
+Gherkin scenarios are used to describe the functional requirements of your software. They should be readable for the team and also for the business people that use the software. Technical id's don't have a place in these scenarios. So what to do when your code requires a technical id?
```

Comment:
Business people klint een beetje vreemd. Ik zou people weglaten en de zin dan aanpassen: They should be readable for the team and for the business that uses the software.

Reply:
Ik zou iets meer vertellen waarom Technical id's geen plek hebben in je scenario's. In dit geval gebruik je Guid's, wat een scenario al heel lelijk maakt. In het geval van een int denk ik dat een developer snel zou zeggen dat het wel leesbaar is. Hoewel id 1 natuurlijk niet functioneel is.

Reply:
Punt maak ik denk ik in 'Refactoring our scenario'. Heb die met nog een reden, naast leesbaarheid uitgebreid.